### PR TITLE
Catch DOMErors not just Errors (fixes #80)

### LINKED ISSUE #80: Support DOMError instance in client/service method reject
Currently the method could support basic error instance in reject that client side could get the error message directly in method reject, but for DOMError the method will be failed in dataclone error. Considering the DOMError is commonly used in our WebAPI, it would be better if bridge could deal with dataclone issue instead of cloned/handled by service itself.

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

File: bridge.js [hunk 1]
```diff
@@ -994,6 +1000,24 @@ Receiver.prototype = {
...
+ *
+ * @param  {*} err
+ * @return {Object|*}
+ */
+function serializeError(err) {
+  switch (err && err.constructor.name) {
+    case 'Error': return { message: err.message };
+    case 'DOMError': return { message: err.message, name: err.name };
+    case 'DOMException': return { message: err.toString() };
+    default: return err;
```
Comment (wilsonpage) [inline bridge.js:1016]: @azasypkin I noticed that if an exception throws inside a Promise chain then the argument that gets passed to `.catch(err => ...` is a `DOMException` which is a little like a `String` but not cloneable.

I'm wondering if we should normalize all errors thrown in Services into a single string in the form of `DOMException`. I still don't know enough about the `DOMError`s thrown by your messaging service to know if this is a good idea.

Comment (azasypkin) [inline bridge.js:1016]: > @azasypkin I noticed that if an exception throws inside a Promise chain then the argument that gets passed to .catch(err => ... is a DOMException which is a little like a String but not cloneable.

Mmm, I'm surprised :) Do you have an example? I believe(**-d**) promise chain should not change error type.

Comment (wilsonpage) [inline bridge.js:1016]: Hmmm perhaps this was just for this particular case. It's tricky to inspect these Error objects when they're inside Workers. Are you still in favour of keeping the Error object structure the same then?

Comment (wilsonpage) [inline bridge.js:1016]: I noticed that there may be other Error types that we are not checking for and could thus throw clone errors at .postMessage(), eg. `ReferenceError`.  Do yo know of anymore, any suggestions on how to cope with this?

**Comment by wilsonpage** [comment]
@azasypkin can you confirm this fixes things for you. Then I will stamp new `v1.0.0`.

**Comment by azasypkin** [comment]
@wilsonpage, sure, will check and get back to you

**Comment by azasypkin** [comment]
@wilsonpage yep, at least for the case Steve described in #80 it works perfectly well now, thanks!

**Comment by wilsonpage** [comment]
@azasypkin and this warrants a 'breaking change' major semver bump?

**Comment by azasypkin** [comment]
@wilsonpage IMO if we follow good practices, then yes since format has changed :) But in this particular case it won't be breaking change for Messages app - we mostly output bridge errors directly to `console.error` and have workaround for `DOMError` in place (`throw error instanceof DOMError ? { name: error.name } : error;`).

**Pull request merged**